### PR TITLE
A few small SYCL 2020 v4 updates for the common interface semantics on Ubuntu

### PR DESCRIPTION
**Pull request by pkeir**

### CODE REVIEW

File: ubuntu-packages.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 clang-format
-enchant
+enchant-2
 git
 python3
 python3-pip
 python3-venv
+python3-sphinx
+python3-sphinxcontrib.spelling
```
Comment (rscohn2) [inline ubuntu-packages.txt:8]: Normally, these would come from here: https://github.com/KhronosGroup/SYCL_Reference/blob/8802b5beb5bb2096c15ad48dda14a2bcfb6be998/requirements.txt#L6
when you do 
```
pip3 install -r requirements.txt
```

I always work in a virtual environment and prefer not to install system packages. Are you OK with not adding these 2 here?

**Review by rscohn2 — COMMENTED** [review]
Thanks for the contributions.

File: ubuntu-packages.txt (hunk 1 above)
Comment (pkeir) [reply ubuntu-packages.txt:8]: Yes, no problem.

File: source/iface/common-byval.rst [hunk 2]
```diff
@@ -9,13 +9,14 @@
 ====================
 
 Types: :ref:`id`, :ref:`range`, :ref:`item`, :ref:`nd_item`,
-:ref:`h_item`, :ref:`group`, and :ref:`nd_range`.
+:ref:`h_item`, :ref:`group`, `sub_group`, and :ref:`nd_range`.
```
Comment (keryell) [inline source/iface/common-byval.rst:12]: Just curious, why `:ref:` everywhere but not on this one?

Comment (rscohn2) [reply source/iface/common-byval.rst:12]: sub_group is not documented yet. I have been documenting sycl 2020 features when I use them so I can add an example.

**Pull request merged**